srml/support/src/storage/mod.rs

```diff
@@ -560,6 +576,109 @@ pub mod unhashed {
+		})
+	}
+
+	/// Return the value of the item in storage under `key`, or the type's default if there is no
+	/// explicit entry.
+	pub fn get_or_default<T: Codec + Sized + Default>(storage_key: &[u8], key: &[u8]) -> T {
+		get(storage_key, key).unwrap_or_else(Default::default)
+	}
+
+	/// Return the value of the item in storage under `key`, or `default_value` if there is no
+	/// explicit entry.
+	pub fn get_or<T: Codec + Sized>(storage_key: &[u8], key: &[u8], default_value: T) -> T {
+		get(storage_key, key).unwrap_or(default_value)
+	}
+
+	/// Return the value of the item in storage under `key`, or `default_value()` if there is no
+	/// explicit entry.
+	pub fn get_or_else<T: Codec + Sized, F: FnOnce() -> T>(storage_key: &[u8], key: &[u8], default_value: F) -> T {
+		get(storage_key, key).unwrap_or_else(default_value)
+	}
+
+	/// Put `value` in storage under `key`.
+	pub fn put<T: Codec>(storage_key: &[u8], key: &[u8], value: &T) {
+		value.using_encoded(|slice| runtime_io::set_child_storage(storage_key, key, slice));
+	}
+
+	/// Remove `key` from storage, returning its value if it had an explicit entry or `None` otherwise.
+	pub fn take<T: Codec + Sized>(storage_key: &[u8], key: &[u8]) -> Option<T> {
+		let r = get(storage_key, key);
+		if r.is_some() {
+			kill(storage_key, key);
+		}
+		r
+	}
+
+	/// Remove `key` from storage, returning its value, or, if there was no explicit entry in storage,
+	/// the default for its type.
+	pub fn take_or_default<T: Codec + Sized + Default>(storage_key: &[u8], key: &[u8]) -> T {
+		take(storage_key, key).unwrap_or_else(Default::default)
+	}
+
+	/// Return the value of the item in storage under `key`, or `default_value` if there is no
+	/// explicit entry. Ensure there is no explicit entry on return.
+	pub fn take_or<T: Codec + Sized>(storage_key: &[u8],key: &[u8], default_value: T) -> T {
+		take(storage_key, key).unwrap_or(default_value)
+	}
+
+	/// Return the value of the item in storage under `key`, or `default_value()` if there is no
+	/// explicit entry. Ensure there is no explicit entry on return.
+	pub fn take_or_else<T: Codec + Sized, F: FnOnce() -> T>(storage_key: &[u8], key: &[u8], default_value: F) -> T {
+		take(storage_key, key).unwrap_or_else(default_value)
+	}
+
+	/// Check to see if `key` has an explicit entry in storage.
+	pub fn exists(storage_key: &[u8], key: &[u8]) -> bool {
+		runtime_io::read_child_storage(storage_key, key, &mut [0;0][..], 0).is_some()
+	}
+
+	/// Remove all `storage_key` key/values 
+	pub fn kill_storage(storage_key: &[u8]) {
+		runtime_io::kill_child_storage(storage_key)
+	}
+
+	/// Ensure `key` has no explicit entry in storage.
+	pub fn kill(storage_key: &[u8], key: &[u8]) {
+		runtime_io::clear_child_storage(storage_key, key);
+	}
+
+	/// Ensure keys with the given `prefix` have no entries in storage.
+	pub fn kill_prefix(storage_key: &[u8], prefix: &[u8]) {
+		runtime_io::clear_child_prefix(storage_key, prefix);
```

Comment:
I probably missed it, but do you mind to explain why we still need prefix for a child storage?